lib/plugins/inspec-streaming-reporter-progress-bar/lib/inspec-streaming-reporter-progress-bar/streaming_reporter.rb

```diff
@@ -9,104 +10,144 @@ module InspecPlugins::StreamingReporterProgressBar
+      control_impact = notification.example.metadata[:impact]
+      set_status_mapping(control_id, 'skipped')
+      show_progress(control_id, title, full_description, control_impact) if control_ended?(control_id)
+    end
+
+    def example_error(notification)
+      control_id = notification.example.metadata[:id]
+      title = notification.example.metadata[:title]
+      full_description = notification.example.metadata[:full_description]
+      control_impact = notification.example.metadata[:impact]
+      set_status_mapping(control_id, 'error')
+      show_progress(control_id, title, full_description, control_impact) if control_ended?(control_id)
+    end
+
+    def example_na(notification)
+      control_id = notification.example.metadata[:id]
+      title = notification.example.metadata[:title]
+      full_description = notification.example.metadata[:full_description]
+      control_impact = notification.example.metadata[:impact]
+      set_status_mapping(control_id, 'na')
+      show_progress(control_id, title, full_description, control_impact) if control_ended?(control_id)
     end
 
     private
 
-    def show_progress(control_id)
+    def show_progress(control_id, title, full_description, control_impact)
       @bar ||= ProgressBar.new(controls_count, :bar, :counter, :percentage)
       sleep 0.1
       @bar.increment!
-      @bar.puts format_it(control_id)
-    rescue Exception => ex
-      raise "Exception in Progress Bar streaming reporter: #{ex}"
+      @bar.puts format_it(control_id, title, full_description, control_impact)
+    rescue Exception => e
+      raise "Exception in Progress Bar streaming reporter: #{e}"
     end
 
-    def format_it(control_id)
-      control_status = if @status_mapping[control_id].include? "failed"
-                         "failed"
-                       elsif @status_mapping[control_id].include? "skipped"
-                         "skipped"
-                       elsif @status_mapping[control_id].include? "passed"
-                         "passed"
+    def format_it(control_id, title, full_description, control_impact)
+      control_status = if full_description.match?(/Source Code Error/)
+                         'error'
+                       elsif control_impact == 0
+                         'na'
+                       elsif @status_mapping[control_id].uniq.include?('failed')
+                         'failed'
+                       elsif @status_mapping[control_id].uniq.include?('skipped') && @status_mapping[control_id].uniq.one?
+                         'skipped'
+                       elsif @status_mapping[control_id].uniq.any? { |val| /"passed"||"skipped"/ =~ val }
+                         'passed'
                        end
 
       indicator = INDICATORS[control_status]
-      message_to_format = ""
-      message_to_format += "#{indicator}  "
-      message_to_format += control_id.to_s.lstrip.force_encoding(Encoding::UTF_8)
```

Comment:
`force_encoding(Encoding::UTF_8)` we should probably keep this, this is a safe check.